test/test.service.getPermissionsForUser.ts

```diff
@@ -0,0 +1,61 @@
+import GetPopulatedPermissionsForCurrentUser from '../src/services/queries/service.getPermissionsForUser';
+import Permission, { Right } from '../src/interfaces/interface.Permission';
+import sinon from 'sinon';
+import chai,{ expect } from 'chai';
+import * as _ from 'lodash';
+import chaiExclude from 'chai-exclude';
+
+chai.use(chaiExclude);
+
+describe('get permissions for current user',()=>{
+
+  let roleModelMock: any;
+  let roleAssignmentModelMock: any;
+  let permissionModelMock: any;
+  let getPopulatedPermissionsForCurrentUser: (user: string|null) => Promise<Permission[]>;
+
+  beforeEach(()=>{
+    roleModelMock = { collection:{ name:'123' } };
+    permissionModelMock = { collection:{ name:'123' } };
+    roleAssignmentModelMock = { aggregate:sinon.stub() };
+    getPopulatedPermissionsForCurrentUser=GetPopulatedPermissionsForCurrentUser(permissionModelMock, roleAssignmentModelMock, roleModelMock);
+
+  });
+
+
+  it('should fill query restriction with given data',async ()=> {
+    const expectedPermission: Permission= {
+      Title:'ReadBlog',
+      Right:Right.read,
```

Comment:
vielleicht sollten wir uns überlegen den enum in einen type umzubauen. Habe zwar ich gemacht aber finde eigentlich einen type sinnvoller hier

Reply:
es gibt mittlerweile auch string enums, dann steht da auch was lesbares